update_doc.py

```diff
@@ -122,15 +126,155 @@ def check_supported_models(check: bool):
+    os.remove("/tmp/server_log.txt")
+
+
+def get_openapi_json():
+    response = requests.get("http://127.0.0.1:3000/api-doc/openapi.json")
+    # error if not 200
+    response.raise_for_status()
+    return response.json()
+
+
+def update_openapi_json(new_data, filename="docs/openapi.json"):
+    with open(filename, "w") as f:
+        json.dump(new_data, f, indent=2)
+
+
+def compare_openapi(old_data, new_data):
+    differences = []
+
+    def compare_recursive(old, new, path=""):
+        if isinstance(old, dict) and isinstance(new, dict):
+            for key in set(old.keys()) | set(new.keys()):
+                new_path = f"{path}.{key}" if path else key
+                if key not in old:
+                    differences.append(f"Added: {new_path}")
+                elif key not in new:
+                    differences.append(f"Removed: {new_path}")
+                else:
+                    compare_recursive(old[key], new[key], new_path)
+        elif old != new:
+            differences.append(f"Changed: {path}")
+
+    compare_recursive(old_data, new_data)
+    return differences
+
+
+def openapi(check: bool):
+    try:
+        server_process, log_file = start_server_and_wait()
 
-    check_cli(args.check)
-    check_supported_models(args.check)
+        try:
+            new_openapi_data = get_openapi_json()
+
+            if check:
+                try:
+                    with open("docs/openapi.json", "r") as f:
+                        old_openapi_data = json.load(f)
+                except FileNotFoundError:
+                    print(
+                        "docs/openapi.json not found. Run without --check to create it."
+                    )
+                    return
+
+                differences = compare_openapi(old_openapi_data, new_openapi_data)
+
+                if differences:
+                    print("The following differences were found:")
+                    for diff in differences:
+                        print(diff)
+                    print(
+                        "Please run the script without --check to update the documentation."
+                    )
+                else:
+                    print("Documentation is up to date.")
+            else:
+                update_openapi_json(new_openapi_data)
+                print("Documentation updated successfully.")
+
+        finally:
+            stop_server(server_process, log_file)
+
+    except TimeoutError as e:
+        print(f"Error: {e}")
+        raise SystemExit(1)
+    except requests.RequestException as e:
+        print(f"Error communicating with the server: {e}")
+        raise SystemExit(1)
+    except json.JSONDecodeError:
+        print("Error: Invalid JSON received from the server")
+        raise SystemExit(1)
+    except Exception as e:
+        print(f"An unexpected error occurred: {e}")
+        raise SystemExit(1)
 
 
 if __name__ == "__main__":
-    main()
+    parser = argparse.ArgumentParser(
```

Comment:
Can we keep the simpler version that just checks everything by default ?
instead just add `check_openapi` function ?

Reply:
yep that makes sense, updated in the latest commits